src/EventListener/FormFieldDcaListener.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the inspiredminds/contao-fieldset-duplication package.
+ *
+ * (c) inspiredminds
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace InspiredMinds\ContaoFieldsetDuplication\EventListener;
+
+use Contao\Controller;
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+
+final class FormFieldDcaListener
+{
+    /** @var ContaoFrameworkInterface */
+    private $framework;
+
+    public function __construct(ContaoFrameworkInterface $framework)
+    {
+        $this->framework = $framework;
+    }
+
+    public function templateOptions(): array
+    {
+        $adapter = $this->framework->getAdapter(Controller::class);
```

Comment:
Is there any benefit from using these adapters, if you are not doing testing?

Reply:
You're able to add tests later on. :-) Probably it's personal preference. I almost always use it so that my codebase uses the same API of the Contao framework no matter if it's test/spec covered or not.

Reply:
Alright :)